packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SVGComponents/VXZoom/VXZoom.js

```diff
@@ -0,0 +1,160 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import { localPoint } from '@vx/event'
+import { Zoom } from '@vx/zoom'
+import ZoomEventLayer from '../ZoomEventLayer'
+
+class VXZoom extends PureComponent {
+  constructor(props) {
+    super(props)
+    const { setOnZoom } = props
+
+    setOnZoom(this.handleToolbarZoom.bind(this))
+
+    this.onDoubleClick = this.onDoubleClick.bind(this)
+    this.onMouseLeave = this.onMouseLeave.bind(this)
+  }
+
+  static zoom = null
+
+  handleToolbarZoom(type) {
+    const doZoom = {
+      'zoomin': this.zoomIn.bind(this),
+      'zoomout': this.zoomOut.bind(this),
+      'zoomto': this.zoomReset.bind(this)
+    }
+
+    if (doZoom[type]) {
+      doZoom[type]()
+    }
+  }
+
+  zoomIn() {
+    if (!this.props.zooming) return
+    const { zoomInValue } = this.props.zoomConfiguration
+    this.zoom.scale({ scaleX: zoomInValue, scaleY: zoomInValue })
+  }
+
+  zoomOut() {
+    if (!this.props.zooming) return
+    const { zoomOutValue } = this.props.zoomConfiguration
+    this.zoom.scale({ scaleX: zoomOutValue, scaleY: zoomOutValue })
+  }
+
+  zoomReset() {
+    if (!this.props.zooming) return
+    this.zoom.reset()
+  }
+
+  zoomToPoint(event, zoomDirection) {
+    const { zoomInValue, zoomOutValue } = this.props.zoomConfiguration
+    const zoomValue = (zoomDirection === 'in') ? zoomInValue : zoomOutValue
+    const point = localPoint(event)
+    this.zoom.scale({ scaleX: zoomValue, scaleY: zoomValue, point })
+  }
+
+  onDoubleClick(event) {
+    if (this.props.zooming) {
+      this.zoomToPoint(event, 'in')
+    } else {
+      event.preventDefault()
+    }
+  }
+
+  onMouseLeave() {
+    if (!this.zoom.isDragging && !this.props.panning) return
+    this.zoom.dragEnd()
+  }
+
+  onWheel(event) {
+    // performance of this is pretty bad
```

Comment:
> performance of this is pretty bad

Now here's a weird one; I was going to disagree with you on this, until I found out what triggers the poor performance.
- when using _a MacBook Pro's trackpad,_ on macOS + Chrome78, if the _dev console is open,_ then zooming via mouse scroll is extremely jaggy, jumping inconsistently between multiple zoom levels per scroll.
- when using _a MacBook Pro's trackpad,_ on macOS + Firefox 70, zooming via mouse scroll isn't very smooth, but better than Chrome with the dev console open.
- on macOS 10 + Safari, zooming via mouse scroll is pretty smooth, which is an affront to everything I understand about browsers.
- when using _an external USB mouse,_ zooming via mouse scroll is mostly smooth on all tested conditions and browsers.
  - (Chrome with the dev console open was slightly slower, but not as janked out as with touchpad + open dev console)

Overall thoughts summarised as an emoji: 😕

Reply:
I've copied your comment into https://github.com/zooniverse/front-end-monorepo/issues/1143#issuecomment-558698182

I think we'll have to make a final decision about supporting mouse wheel at some point before finishing this. I'm not 100% on supporting it yet.